### PR TITLE
Reorder Award Table Columns

### PR DESCRIPTION
**Pull request by ebdabbs**
- From the dropdown, user can drag and drop visible columns to change their order in the table

### CODE REVIEW

**Comment by geoffgeiling** [comment]
I have reviewed the design and it sufficiently meets the acceptance criteria... and looks AMAZING!!

File: src/js/components/search/table/ResultsColumnVisibleOption.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            return;
+        }
+
+        // Dragging upwards
+        if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+            return;
+        }
+
+        // Perform the action
+        props.moveColumn(dragIndex, hoverIndex);
```
Comment (kevinli-work) [inline src/js/components/search/table/ResultsColumnVisibleOption.jsx]: Is it worth differentiating between a drag/hover event and a drop event? As currently written, the table updates (and triggers an API call) after every hover event, so dragging the first column to the end triggers a bunch of unnecessary API calls.

File: src/js/redux/reducers/search/columnVisibilityReducer.js [hunk 2]
```diff
@@ -14,12 +14,12 @@ export const VisibilityRecord = Record(initialState);
 const columnVisibilityReducer = (state = initialState, action) => {
     switch (action.type) {
         case `TOGGLE_COLUMN_VISIBILITY`: {
+            const updatedVisible = ColumnVisibilityFuncs.toggleItem(
+                state[action.tableType].visibleColumns, action.column);
+            const updatedHidden = ColumnVisibilityFuncs.toggleItem(
+                state[action.tableType].hiddenColumns, action.column);
             switch (action.tableType) {
```
Comment (ebdabbs) [inline src/js/redux/reducers/search/columnVisibilityReducer.js]: Replaced switch statement with computed properties

File: src/js/redux/reducers/search/columnVisibilityReducer.js [hunk 3]
```diff
@@ -99,6 +83,78 @@ const columnVisibilityReducer = (state = initialState, action) => {
...
             }
         }
+        case 'REORDER_COLUMNS': {
+            const tableType = action.tableType;
+
+            const updatedVisible = ColumnVisibilityFuncs.spliceColumnOrder(
+                state[`${tableType}`].visibleColumns,
+                action.dragIndex, action.hoverIndex);
+
+            switch (action.tableType) {
```
Comment (ebdabbs) [inline src/js/redux/reducers/search/columnVisibilityReducer.js]: Replaced switch statement with computed properties

File: src/js/redux/reducers/search/filters/ColumnVisibilityFunctions.js [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
     if (selected.includes(item)) {
         return selected.delete(item);
     }
 
     return selected.add(item);
 };
+
+export const spliceColumnOrder = (currentSet, dragIndex, hoverIndex) => {
+    // Convert the "before" Ordered Set to an Array
+    const visibleArray = Array.from(currentSet);
```
Comment (ebdabbs) [inline src/js/redux/reducers/search/filters/ColumnVisibilityFunctions.js]: Replaced Array.from() with toArray()

File: src/js/components/search/table/ResultsColumnVisibleOption.jsx (hunk 1 above)
Comment (ebdabbs) [inline src/js/components/search/table/ResultsColumnVisibleOption.jsx]: Tried a few options but I think the way the order is stored will have to be structured differently to maintain the shuffling animation when dragging a column.

**Pull request merged**